tests/test_second_kick.py

```diff
@@ -0,0 +1,271 @@
+    kcrit=0.  This is nontrivial since the SecondKick structure function is numerically integrated,
+    while the vK structure function is evaluated analytically.
+    """
+    lam = 500.0
+    r0 = 0.2
+    L0 = 30.0
+    diam = 8.36
+    obscuration = 0.61
+
+    sk = galsim.SecondKick(lam, r0, diam, obscuration, L0, kcrit=0.0)
+    vk = galsim.VonKarman(lam, r0, L0=L0)
+
+    for rho in [0.01, 0.03, 0.1, 0.3, 1.0, 3.0]:
+        sksf = sk._structure_function(rho)
+        vksf = vk._structure_function(rho)
+        np.testing.assert_allclose(sksf, vksf, rtol=1e-6)
+
+
+@timer
+def test_limiting_cases():
+    """SecondKick has some two interesting limiting cases.
+    A) When kcrit = 0, SecondKick = Convolve(Airy, VonKarman).
+    B) When kcrit = inf, SecondKick = Airy
+    Test these.
+    """
+    lam = 500.0
+    r0 = 0.2
+    L0 = 30.0
+    diam = 8.36
+    obscuration = 0.61
+
+    # First kcrit=0
+    sk = galsim.SecondKick(lam, r0, diam, obscuration, L0, kcrit=0.0)
+    limiting_case = galsim.Convolve(
+        galsim.VonKarman(lam, r0, L0=L0),
+        galsim.Airy(lam=lam, diam=diam, obscuration=obscuration)
+    )
+    print(sk.stepk, sk.maxk)
+    print(limiting_case.stepk, limiting_case.maxk)
+
+    for k in [0.0, 0.1, 0.3, 1.0, 3.0, 10.0, 30.0]:
+        np.testing.assert_allclose(
+            sk.kValue(0, k).real,
+            limiting_case.kValue(0, k).real,
+            rtol=1e-7,
+            atol=1e-8)
+
+    # kcrit=inf
+    sk = galsim.SecondKick(lam, r0, diam, obscuration, L0, kcrit=np.inf)
+    limiting_case = galsim.Airy(lam=lam, diam=diam, obscuration=obscuration)
+
+    for k in [0.0, 0.1, 0.3, 1.0, 3.0, 10.0, 30.0]:
+        np.testing.assert_allclose(
+            sk.kValue(0, k).real,
+            limiting_case.kValue(0, k).real,
+            rtol=1e-3,
+            atol=1e-4)
+    # Check half_light_radius agrees.  Only implemented for obscuration=0, and somewhat
+    # approximate for second kick.
+    airy = galsim.Airy(lam=lam, diam=diam)
+    sk = galsim.SecondKick(lam=lam, r0=r0, diam=diam, kcrit=np.inf)
+    assert np.abs(sk.half_light_radius/airy.half_light_radius - 1.0) < 1e-2
+
+
+@timer
+def test_sf_lut(slow=False):
+    """Test the suitability of the lookup table used to store the structure function by comparing
+    results of xValue and kValue both with and without using the LUT.
+    """
+    lam = 500.0
+    r0 = 0.2
+    L0 = 30.0
+    diam = 8.36
+    obscuration = 0.61
+    kcrit = 2*np.pi/r0
+
+    sk = galsim.SecondKick(lam, r0, diam, obscuration, L0, 0.5*kcrit)
+
+    print("Testing kValue")
+    for k in [0.0, 0.01, 0.03, 0.1, 0.3, 1.0, 3.0, 10.0, 30.0, 100.0, 300.0, 600.0]:
+        print()
+        print("k = {}".format(k))
+        print(sk.kValue(0, k).real)
+        print(sk._sbp.kValueSlow(k))
+        np.testing.assert_allclose(
+            sk.kValue(0, k).real,   # Uses LUT
+            sk._sbp.kValueSlow(k),  # No LUT
+            rtol=1e-5,
+            atol=1e-5
+        )
```

Comment:
I think this may be useful at least until everything else here has converged (I did comment it out for now).  If you want to remove it at that point, that's fine with me.